lib/EntryDownloader.js

```diff
@@ -34,23 +42,71 @@ var EntryDownloader = (function () {
         }
     };
 
+    var checkTranscodingProfileChange = function(entryId, flavorsPromiseObj) {
+        var entryPath = persistenceFormat.getEntryFullPath(entryId);
+        var reg = new RegExp("(?:" + entryId + "\\/)([0-9]*)\\b", "i");
+        return qio.listDirectoryTree(entryPath)
+            .then(function(result) {
+                return _.chain(result)
+                    .filter(function(d) {
+                        return d.match(reg) !== null;
+                    })
+                    .map(function(d) {
+                        return d.match(reg)[1];
+                    })
+                    .uniq(function(f) {
+                        return f;
+                    })
+                    .value();
+            })
+            .then(function(result) {
+                var playingFlavorsString = flavorsPromiseObj.flavorsList.join();
+                if (result.join() !== playingFlavorsString) {
+                    logger.info("Transcoding profile changed, cleaning entry " + entryId + "folder");
+                    return fsUtils.cleanFolder(entryPath);
+                }
+            })
+            .then(function() {
+                return flavorsPromiseObj;
+            })
+            .catch(function(error) {
+                logger.error("Error cleaning folder because of transcoding profile change: " + error);
+            });
+    };
+
+    var savePlaylist = function (flavorsPromiseObj, entryId) {
+        // Get the destination to save the manifest
+        var destPath = path.join(config.get('rootFolderPath'), entryId, persistenceFormat.getMasterManifestName());
+        //return qfs.write(destPath, manifest);
+        return fsUtils.writeFileAtomically(destPath, flavorsPromiseObj.playlist)
+            .then(function() {
+                return flavorsPromiseObj.flavorsObj
+            })
+    };
+
     EntryDownloader.prototype.start = function () {
         var that = this;
         logger.info("Entry start requested for entry " + that.entryId);
-        var persistenceFormat = require('./../common/PersistenceFormat');
         logger.info("Starting entry downloader for entryId: " + that.entryId + "\n");
         return that.masterManifestGenerator.getAllFlavors()
             .then(function(flavors) {
+               return (!that.isNewSession) ? checkTranscodingProfileChange(that.entryId, flavors) : flavors;
```

Comment:
maybe it's better to put the isNewSession check into the  checkTranscodingProfileChange?